Lib/code.py

```diff
@@ -132,14 +133,18 @@ def showtraceback(self):
         """
         try:
             typ, value, tb = sys.exc_info()
-            self._showtraceback(typ, value, tb.tb_next)
+            self._showtraceback(typ, value, tb.tb_next, "")
         finally:
             typ = value = tb = None
 
-    def _showtraceback(self, typ, value, tb):
+    def _showtraceback(self, typ, value, tb, source):
         sys.last_type = typ
         sys.last_traceback = tb
-        sys.last_exc = sys.last_value = value = value.with_traceback(tb)
+        value = value.with_traceback(tb)
+        # Set the line of text that the exception refers to
+        if source and typ is SyntaxError and not value.text:
+            value.text = source.splitlines()[value.lineno - 1]
```

Comment:
This can fail if somehow source is not big enough and we don't want to raise in this case. As a precaution, please add another condition or another guard to ensure that there are enough lines in source to do this indexing